src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo).append(System.lineSeparator());
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+        for (String name : names) {
+            int salary = 0;
+            for (String entry : data) {
+                String[] parts = entry.split(" ");
+                String recordDateStr = parts[0];
```

Comment:
Any magic numbers should be constants

see https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info